### PR TITLE
Basic support for Xirrus Wireless Access Points.

### PR DESCRIPTION
**Pull request by Tatermen**
Add basic support for Xirrus Wireless Access Points.

This may need further work - the wifi polling code only appears to support graphing the station count for two radios - Xirrus APs can have upto 16, but I only have access to a two radio model at present.

### CODE REVIEW

File: includes/discovery/os/xirrus.inc.php [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+if(!$os) {
+    if (strstr($sysDescr, 'Xirrus')) {
+        $os = 'xirrus';
+    }
+}
+
+?>
```
Comment (murrant) [inline includes/discovery/os/xirrus.inc.php]: Please remove the `?>`

Comment (Tatermen) [inline includes/discovery/os/xirrus.inc.php]: Took me two attempts, but it's gone :)

File: includes/polling/wireless/xirrus.inc.php [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    if (!is_file($rrd_filename)) {
+        rrdtool_create($rrd_filename, " --step 300 DS:rssi:GAUGE:600:-150:0".$config['rrd_rra']);
+    }
+    rrdtool_update($rrd_filename, array('rssi'=>$rssi[$idx]['realtimeMonitorAverageRSSI']));
+
+}
+// cleanup
+unset($rrd_filename); unset($radios); unset($rssi); unset($radioName);
+$graphs['xirrus_rssi'] = TRUE;
+?>
```
Comment (f0o) [inline includes/polling/wireless/xirrus.inc.php]: Please remove this line

**Comment by geordish** [comment]
I think this needs to be ArrayOS as the OS

**Comment by stevewalsh** [comment]
I have snmp read access to a bunch of Xirrus arrays for a conference until 2 June. Please let me know what sort of information you need, and I'll run snmpwalks against the devices I have access to.

**Comment by Tatermen** [comment]
You're right - I've changed the OS name to "Xirrus ArrayOS" (internally xirrus_aos) which should prevent any conflicts.

I've written it to walk the following OIDs, to get the majority of the stats:

```
XIRRUS-MIB::realtimeMonitorIfaceName
XIRRUS-MIB::realtimeMonitorAverageRSSI
XIRRUS-MIB::realtimeMonitorAverageDataRate
XIRRUS-MIB::realtimeMonitorNoiseFloor
```

On my Xirrus XR620 which only has two radios, these return tables with an OID for each radio:

```
XIRRUS-MIB::realtimeMonitorIfaceName.1 = STRING: iap1
XIRRUS-MIB::realtimeMonitorIfaceName.2 = STRING: iap2
XIRRUS-MIB::realtimeMonitorAverageRSSI.1 = INTEGER: -93
XIRRUS-MIB::realtimeMonitorAverageRSSI.2 = INTEGER: 0
XIRRUS-MIB::realtimeMonitorAverageDataRate.1 = INTEGER: 59
XIRRUS-MIB::realtimeMonitorAverageDataRate.2 = INTEGER: 0
XIRRUS-MIB::realtimeMonitorNoiseFloor.1 = INTEGER: -102
XIRRUS-MIB::realtimeMonitorNoiseFloor.2 = INTEGER: 0
```

So I've written the code on the assumption that further radios will show up as additional OIDs, eg:

`
XIRRUS-MIB::realtimeMonitorIfaceName.3 = STRING: iap3
XIRRUS-MIB::realtimeMonitorIfaceName.4 = STRING: iap4
`

if this is correct, then I'm confident the code I've written should correctly poll the additional radios. Station associations work in a similar manner, where there is a table of OIDs of associations' MAC addresses, IPs, device names etc, as well as what radio they are connected to:

```
XIRRUS-MIB::stationAssociationIAP.1 = STRING: iap1
XIRRUS-MIB::stationAssociationIAP.2 = STRING: iap2
XIRRUS-MIB::stationAssociationIAP.3 = STRING: iap1
XIRRUS-MIB::stationAssociationIAP.4 = STRING: iap1
```

So again I'm assuming that additional radios will be called iap3, iap4, iap5 and so on. An issue with doing this may be that it may take a long time to poll an AP that has hundreds or thousands of users on it, but there does not appear to be any OIDs that just list a count of stations per radio.

**Comment by laf** [comment]
How likely is it to have 100's / 1000's of users on each AP? We wouldn't want to slow polling down for people who heavily use these.

**Comment by Tatermen** [comment]
Not too likely I would think. To the best of my knowledge the ones that support thousands of users costs $40-50,000. 

I could add a configuration flag to disable the station/user polling if you think it could be of concern, something like `$config['xirrus_disable_stations']` ?

**Comment by laf** [comment]
I'd say that would be best - just in case.

**Comment by geordish** [comment]
You should probably set a default value for $config['xirrus_disable_stations'] in includes/defaults.inc.php

**Comment by Tatermen** [comment]
Done. I didn't realize that existed,

**Pull request merged**